### PR TITLE
Rest api versioning

### PR DESCRIPTION
**Pull request by kretep**
Updates to configuration to support versioning in the rest api. Requires the following PR's to be merged:
https://github.com/transmart/transmartApp/pull/191
https://github.com/transmart/transmart-data/pull/150

### CODE REVIEW

File: src/app/config.json [hunk 1]
```diff
@@ -1,30 +1,66 @@
 {
     "dev": {
-        "isTesting": false,
-        "masterEndpointConfig": {
+        "IS_TESTING": false,
+        "MASTER_ENDPOINT_CONFIG": {
```
Comment (rnugraha) [inline src/app/config.json:4]: Why are we introducing new naming convention?

Comment (kretep) [reply src/app/config.json:4]: Did we have any naming convention before then?

Comment (rnugraha) [reply src/app/config.json:4]: No. And so far in our projects we do it in cameCase before. My personal objection on this is all caps make things like constants. And your argument to introducing this is?

File: src/app/config.json [hunk 2]
```diff
@@ -1,30 +1,66 @@
...
+                "title": "transmart-gb",
+                "url": "http://transmart-gb.thehyve.net/transmart/v1",
+                "isOAuth": "true"
+            },
+            {
+                "title": "transmart-test-translocation",
+                "url": "http://transmart-test-translocation.thehyve.net/transmart/v1",
+                "isOAuth": "true"
+            }
+        ]
```
Comment (rnugraha) [inline src/app/config.json:42]: now it the keys are look so inconsistent with the mixed of naming style. And again if we introduce mixed naming styling will raise questions  on what and which level should be on this or that style.

If you want to see them as constants in Angular world, maybe it's better to translate it with angular.constant.

```
angular.module('app', []);

app.constant('MOVIE_TITLE', 'The Matrix');

.controller('MyController', function (MOVIE_TITLE) {
  expect(MOVIE_TITLE).toEqual('The Matrix');
});
```

**Pull request merged**